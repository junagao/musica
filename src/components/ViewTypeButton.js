import React from 'react';
import PropTypes from 'prop-types';

import './ViewTypeButton.scss';

const ViewTypeButton = ({ onChangeView, viewType, icon }) => (
  <button
    type="button"
    className="view-btn"
    onClick={() => onChangeView(viewType)}
  >
    {icon}
  </button>
);

ViewTypeButton.propTypes = {
  onChangeView: PropTypes.func.isRequired,
  viewType: PropTypes.string.isRequired,
  icon: PropTypes.element.isRequired,
};

export default ViewTypeButton;
