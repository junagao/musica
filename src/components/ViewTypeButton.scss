@import '../variables';

.view-btn {
  color: $color-lighter;
  background-color: transparent;
  margin-left: 0.75rem;
  font-size: 1.5rem;
  cursor: pointer;
  outline: none;
  border: 0;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  transition: all ease-out 0.3s;
  &:hover,
  &:focus {
    color: $color-darker;
    transition: all ease-in 0.3s;
  }
}
