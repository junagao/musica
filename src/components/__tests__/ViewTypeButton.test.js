import React from 'react';
import { shallow } from 'enzyme';
import { FaThList } from 'react-icons/fa';

import ViewTypeButton from 'components/ViewTypeButton';

describe('ViewTypeButton', () => {
  let wrapper;
  const onChangeViewMock = jest.fn();
  const viewTypeMock = 'list';
  const iconMock = <FaThList />;

  beforeEach(() => {
    wrapper = shallow(
      <ViewTypeButton
        onChangeView={onChangeViewMock}
        viewType={viewTypeMock}
        icon={iconMock}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one button element of type `button` and className `view-btn`', () => {
    expect(wrapper.find('button').length).toEqual(1);
    expect(wrapper.find('button').prop('type')).toEqual('button');
    expect(wrapper.find('button').hasClass('view-btn')).toBeTruthy();
  });

  it('should render an icon `<FaThList />`', () => {
    expect(wrapper.find('button').text()).toEqual('<FaThList />');
  });

  it('should test click event on button and pass `list` to onChangeView()', () => {
    wrapper.find('button').simulate('click');
    expect(onChangeViewMock.mock.calls[0][0]).toEqual('list');
    expect(onChangeViewMock.mock.calls.length).toEqual(1);
  });
});
