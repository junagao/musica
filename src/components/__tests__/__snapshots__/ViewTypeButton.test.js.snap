// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTypeButton should render correctly given the required props 1`] = `
<button
  className="view-btn"
  onClick={[Function]}
  type="button"
>
  <FaThList />
</button>
`;
